src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1147,6 +1193,19 @@ private AuthenticationResult acquireTokenAfterValidation(CallbackHandler callbac
             AuthenticationResult result = null;
             request.setVersion(getVersionName());
             request.setBrokerAccountName(request.getLoginHint());
+            
+            //check if the redirectUri is valid
+            try 
```

Comment:
i think we should move the try/catch inside the redirect uri check method itself.